feat(nitrorole): add a command for showing server settings

cool-aid-man opened the pull request:
Hey Jack,

Added a settings sub-command to the `NitroRole` cog. Makes it easier to view the cog config.
So, thought of pr-ing the changes. Let me know if this is okay!

- Checking the cog enable status by checking if the channel is set else none.
- Aliases `show`, `showsettings`, `setting`

cool-aid-man committed:
[NitroRole] Added `settings` - making it easier to view the config

cool-aid-man committed:
Adjust aliases

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,42 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        # Check if the channel is set, implying the cog is enabled
+        is_enabled = "Enabled" if channel_id else "Disabled"
+
+        role_mention = f"<@&{role}>" if role else "None"
```
```suggestion
        role_mention = role.mention if role is not None else "None"
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,42 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        is_enabled = "Enabled" if channel_id else "Disabled"
+
+        role_mention = f"<@&{role}>" if role else "None"
+        channel = guild.get_channel(channel_id)
```
Fix typing issue
```suggestion
        channel = channel_id and guild.get_channel(channel_id)
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,42 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+
+        role_mention = f"<@&{role}>" if role else "None"
+        channel = guild.get_channel(channel_id)
+        channel_mention = channel.mention if channel else "None"
```
I have a stylistic preference for using `is None` over truthiness.
```suggestion
        channel_mention = channel.mention if channel is not None else "None"
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -454,3 +490,4 @@ async def maybe_announce(
                 guild.id,
             )
             return
+
```
```suggestion
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,42 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        unassign_status = "Enabled" if unassign_on_boost_end else "Disabled"
+        
+        description = (
+            f"**Auto-assign Role:** {role_mention}\n"
+            f"**Boost Message Channel:** {channel_mention}\n"
+            f"**Unassign Role on Boost End:** {unassign_status}\n"
+            f"**Cog Status:** {is_enabled}"
```
Could you use
https://github.com/Jackenmen/JackCogs/blob/b8a3cab41704b934924f57c64db29d74f4174a75/linkwarner/linkwarner.py#L115-L149
as inspiration for how to format the message? Specifically, I'd like to avoid embeds, only capitalize first word of each setting name, and use Yes/No rather than Enabled/Disabled.

Jackenmen commented:
Also, if you happen to be interested, PRs for equivalent commands being added to some of the other cogs of mine (autogist, banmessage, shell, voicetools, maybe modroles could use one) would be welcome as well.

cool-aid-man committed:
[NitroRole] Changes

- No embed. Sending the config in plain text using a bit of text formatting to make it look good.
- Changed Enabled/Disabled for `Unassign role on boost end` to `Yes`/`No`
  - Kept the `Cog status` to `Enabled/Disabled` - as it doesn't sound good with: *'Cog Status: Yes'* (Cog status: Enabled/Disabled)
- Formatted the role to be mentionable in:
`        await ctx.send(f"Nitro boosters will now be assigned {role.name} role.")`
to
`        await ctx.send(f"Nitro boosters will now be assigned {role.mention} role.")`
- Preferred using `is None` as style for `channel_mention`
- Added `channel_id and ` as requested for typing issue
- Cleaner formatting for: `role_mention = role.mention if role is not None else "None"`

Let me know if i've missed anything

cool-aid-man committed:
new line?

cool-aid-man committed:
Revert "new line?"

This reverts commit c0cbfa662fcc21f74fe3524cc03f3d2e9d7af6d3.

cool-aid-man commented:
Alrighty, I made the changes as you pointed out

Here's the change log:
- No embed. Sending the config in plain text using a bit of text formatting to make it look good.
- Changed Enabled/Disabled for: `Unassign role on boost end` to `Yes`/`No`
  - Kept the `Cog status` to `Enabled/Disabled` - as it doesn't sound good with: *'Cog Status: Yes'* (Cog status: Enabled/Disabled)
- Formatted the role to be mentionable in:
`        await ctx.send(f"Nitro boosters will now be assigned {role.name} role.")`
to
`        await ctx.send(f"Nitro boosters will now be assigned {role.mention} role.")`
- Preferred using `is None` as style for `channel_mention`
- Added `channel_id and ` as requested for typing issue
- Cleaner formatting for: `role_mention = role.mention if role is not None else "None"`

Let me know if I've missed anything

> Also, if you happen to be interested, PRs for equivalent commands being added to some of the other cogs of mine (autogist, banmessage, shell, voicetools, maybe modroles could use one) would be welcome as well.

And Sure! I will pr the changes to the cogs one by one

Jackenmen committed:
Update nitrorole.py

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,36 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+            f"- **Auto-assign role:** {role_mention}\n"
+            f"- **Boost message channel:** {channel_mention}\n"
+            f"- **Unassign role on boost end:** {unassign_status}\n"
+            f"- **Cog status:** {is_enabled}"
```
Move this to top and rename to "Enabled"
```suggestion
            f"- **Enabled:** {is_enabled}"
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,36 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+
+        role = guild.get_role(role_id) if role_id else None
+        role_mention = role.mention if role is not None else "None"
+        channel = channel_id and guild.get_channel(channel_id) if channel_id else None
```
```suggestion
        channel = channel_id and guild.get_channel(channel_id)
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,36 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        # Check if the channel is set, implying the cog is enabled
+        is_enabled = "Enabled" if channel_id else "Disabled"
+
+        role = guild.get_role(role_id) if role_id else None
```
```suggestion
        role = role_id and guild.get_role(role_id)
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,36 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        role = guild.get_role(role_id) if role_id else None
+        role_mention = role.mention if role is not None else "None"
+        channel = channel_id and guild.get_channel(channel_id) if channel_id else None
+        channel_mention = channel.mention if channel is not None else "None"
```
```suggestion
        channel_mention = channel.mention if channel is not None else "*None*"
```

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,36 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        is_enabled = "Enabled" if channel_id else "Disabled"
+
+        role = guild.get_role(role_id) if role_id else None
+        role_mention = role.mention if role is not None else "None"
```
```suggestion
        role_mention = role.mention if role is not None else "*None*"
```

Jackenmen committed:
Apply suggestions from code review

Jackenmen committed:
Move "Enabled" to top

Jackenmen committed:
Fix style

Jackenmen committed:
Fix typing issue

cool-aid-man commented:
Oh, I actually did the same on my local machine lol i.e.: 
```py
role_mention = role.mention if role is not None else "None"
channel = guild.get_channel(channel_id) if channel_id is not None else None
``` 
(Without making the italic ofc)
but since you specifically asked to use `channel_id and ` so I went with it :)

Is there anything else left to be done here!

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,35 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+        unassign_status = "Yes" if unassign_on_boost_end else "No"
+
+        await ctx.send(
+            "## __**NitroRole's Guild Settings**__\n\n"
```
```suggestion
            "## __**NitroRole's Server Settings**__\n\n"
```

Jackenmen committed:
Update nitrorole/nitrorole.py

Jackenmen commented on nitrorole/nitrorole.py:
```diff
@@ -317,6 +317,35 @@ async def cog_disabled_in_guild(self, guild: Optional[discord.Guild]) -> bool:
+            "## __**NitroRole's Server Settings**__\n\n"
+            ">>> "
+            f"- **Enabled:** {is_enabled}\n"
+            f"- **Auto-assign role:** {role_mention}\n"
```
```suggestion
            f"- **Role to auto-assign:** {role_mention}\n"
```

Jackenmen committed:
Update nitrorole/nitrorole.py

Jackenmen commented:
Thanks for contributing :)

> Oh, I actually did the same on my local machine lol i.e.:

I'd prefer the `channel_id and` form but I forgot that this wouldn't work properly since the int type can evaluate to False when 0 so yeah, I had to go back to that.
